### PR TITLE
proj: correct CMake arg for shared build with proj older than 7.0.0

### PR DESCRIPTION
**Pull request by johnwparent**

### CODE REVIEW

File: var/spack/repos/builtin/packages/proj/package.py [hunk 1]
```diff
@@ -128,6 +128,7 @@ def install_datum_grids(self):
 
 class CMakeBuilder(BaseBuilder, cmake.CMakeBuilder):
     def cmake_args(self):
+        shared_arg = "BUILD_SHARED_LIBS" if self.spec.satisfies("@7:") else "BUILD_LIBPROJ_SHARED"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/proj/package.py:131]: You don't actually use this variable...

File: var/spack/repos/builtin/packages/proj/package.py [hunk 2]
```diff
@@ -128,10 +128,11 @@ def install_datum_grids(self):
...
         args = [
             self.define_from_variant("ENABLE_TIFF", "tiff"),
             self.define_from_variant("ENABLE_CURL", "curl"),
-            self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
+            self.define_from_variant(f"{shared_arg}", "shared"),
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/proj/package.py]:
```suggestion
            self.define_from_variant(shared_arg, "shared"),
```
It's already a string.

Comment (johnwparent) [reply var/spack/repos/builtin/packages/proj/package.py]: Right

File: var/spack/repos/builtin/packages/proj/package.py (hunk 1 above)
Comment (johnwparent) [reply var/spack/repos/builtin/packages/proj/package.py:131]: Thanks for catching this, the change where I use this ended up in another branch on rebase, so you've fixed two mistakes.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**